src/modules/todoLogics.js

```diff
@@ -0,0 +1,82 @@
+import variables from './variables';
+
+const myVar = variables();
+
+const save = () => {
+  localStorage.setItem(myVar.LOCAL_STORAGE_PROJECTS_KEY, JSON.stringify(myVar.projectList));
+  localStorage.setItem(myVar.LOCAL_STORAGE_SELECTED_PROJECT_ID_KEY, myVar.selectedProjectId);
+};
+
+const renderProjects = () => {
+  myVar.projectList.forEach(project => {
+    const listElement = document.createElement('li');
+    listElement.classList.add('list-item');
+    listElement.dataset.listId = project.id;
+    listElement.innerText = project.title;
+    if (project.id === myVar.selectedProjectId) {
+      listElement.classList.add('active-list-item');
+    }
+    myVar.projectListContainer.appendChild(listElement);
+  });
+};
+
+const clearElement = (element) => {
+  while (element.firstChild) {
+    element.removeChild(element.firstChild);
+  }
+};
+
+const renderToDosCountFor = (selectedProject) => {
+  const incompleteToDosCount = selectedProject.toDoList.filter(todo => !todo.complete).length;
+  const taskString = incompleteToDosCount === 1 ? 'task' : 'tasks';
+  myVar.toDosCountElement.innerText = `${incompleteToDosCount} ${taskString} remaining`;
+};
+
+const renderToDoListFor = (selectedProjectId) => {
+  selectedProjectId.toDoList.forEach(todo => {
+    const toDoElement = document.importNode(myVar.toDoTemplate.content, true);
+    const detailsElement = toDoElement.querySelector('details');
+    detailsElement.id = todo.id;
+    const descriptionElement = toDoElement.querySelector('.description');
+    descriptionElement.innerText = todo.description;
+
+    const dueDateElement = toDoElement.querySelector('.due-date');
+    dueDateElement.innerText = todo.dueDate;
+
+    const priorityElement = toDoElement.querySelector('.priority');
+    priorityElement.innerText = todo.priority;
+
+
+    const checkbox = toDoElement.querySelector('.checkbox');
+    // checkbox.id = todo.id;
```

Comment:
- [ ] Comments: It’s better to delete commented piece of codes. If you need to review a previous version of your code, use git history
![](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/no_inline_comments.png)